threading/smartptrs.nim

```diff
@@ -93,7 +93,7 @@ proc decr[T](p: SharedPtr[T]) {.inline.} =
   if p.val != nil:
     # this `fetchSub` returns current val then subs
     # so count == 0 means we're the last
-    if p.val.counter.fetchSub(1, Release) == 0:
+    if p.val.counter.fetchSub(1, AcqRel) == 0:
```

Comment:
This causes a *bunch* of data-races with "Release".

Those **might** be false positives as it should be categorically impossible for anyone to still be holding a pointer to SmartPtr after it gets deallocated, but this change her provides certainty for that.

Given that reasoning about this is hard enough as it is I prefer trusting in tsan here.